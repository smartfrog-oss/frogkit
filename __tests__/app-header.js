import { mount, shallow } from 'vue-test-utils'
import appHeader from '@/components/app-header'

const navigations = [
  { label: 'All cameras', url: 'https://app.smartfrog.com/de-de/' },
  { label: 'My account', url: 'https://app.smartfrog.com/de-de/account/overview' },
  { label: 'Shop', url: 'https://www.smartfrog.com/de-de/shop' },
]

const languages = [
  { code: 'de', name: 'Deutsch' },
  { code: 'en', name: 'English' }
]

const data = {
  navigations: navigations,
  languages: languages,
  username: 'john.doe@lorem.com',
  activeLang: 'en'
}

describe('Logged-in Header component', () => {
  it('Should render component and match snapshot', () => {
    const wrapper = mount(appHeader, { propsData: data })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should emit logout event when logout method is called', () => {
    const cmp = shallow(appHeader, {
      propsData: data
    })
    const stub = jest.fn()
    cmp.vm.$on('logout', stub)
    cmp.vm.logout()
    expect(stub).toBeCalled()
  })

  it('Should emit changeLanguage event when changeLang method is called', () => {
    const cmp = shallow(appHeader, {
      propsData: data
    })
    const stub = jest.fn()
    cmp.vm.$on('changeLanguage', stub)
    cmp.vm.changeLang('de')
    expect(stub).toBeCalledWith('de')
  })
})
