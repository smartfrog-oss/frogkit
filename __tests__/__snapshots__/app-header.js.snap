// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logged-in Header component Should render component and match snapshot 1`] = `
<flex>
  <grid>
    <col xs12="">
  </grid>
</flex>
`;
